packages/cloudflare/src/cli/build/patches/to-investigate/wrangler-deps.ts

```diff
@@ -67,3 +72,116 @@ function getDistPath(config: Config): string {
 
   throw new Error("Unexpected error: unable to detect the node_modules/next/dist directory");
 }
+
+/**
+ * `react-dom` v>=19 has a `server.edge` export: https://github.com/facebook/react/blob/a160102f3/packages/react-dom/package.json#L79
+ * but version of `react-dom` <= 18 do not have this export but have a `server.browser` export instead: https://github.com/facebook/react/blob/8a015b68/packages/react-dom/package.json#L49
+ *
+ * Next.js also try-catches importing the `server.edge` export:
+ *  https://github.com/vercel/next.js/blob/6784575/packages/next/src/server/ReactDOMServerPages.js
+ *
+ * The issue here is that in the `.next/standalone/node_modules/next/dist/compiled/next-server/pages.runtime.prod.js`
+ * file for whatever reason there is a non `try-catch`ed require for the `server.edge` export
+ *
+ * This functions fixes this issue by wrapping the require in a try-catch block in the same way Next.js does it
+ * (note: this will make the build succeed but doesn't guarantee that everything will necessarily work at runtime since
+ * it's not clear what code and how might be rely on this require call)
+ *
+ */
+function patchRequireReactDomServerEdge(config: Config) {
+  const distPath = getDistPath(config);
+
+  // Patch .next/standalone/node_modules/next/dist/compiled/next-server/pages.runtime.prod.js
+  const pagesRuntimeFile = join(distPath, "compiled", "next-server", "pages.runtime.prod.js");
+
+  const code = readFileSync(pagesRuntimeFile, "utf-8");
+  const file = tsParseFile(code);
+
+  // we need to update this function: `e=>{"use strict";e.exports=require("react-dom/server.edge")}`
+  file.getDescendantsOfKind(ts.SyntaxKind.ArrowFunction).forEach((arrowFunction) => {
+    // the function has a single parameter
+    const p = arrowFunction.getParameters();
+    if (p.length !== 1) {
+      return;
+    }
+    const parameterName = p[0]!.getName();
+    const bodyChildren = arrowFunction.getBody().getChildren();
+    if (
+      bodyChildren.length !== 3 ||
+      bodyChildren[0]!.getFullText() !== "{" ||
+      bodyChildren[2]!.getFullText() !== "}"
+    ) {
```

Comment:
nit: I find the following easier to read

```suggestion
    if (!(
      bodyChildren.length === 3 &&
      bodyChildren[0]!.getFullText() === "{" &&
      bodyChildren[2]!.getFullText() === "}"
    )) {
```

Reply:
I personally find my version easier to read, but I'm happy to use your version if that's what you prefer 🙂